### PR TITLE
Prevent changelog entries from being inserted to the end of file

### LINKED ISSUE #6: New changelog entries are added to the bottom of the file
Ref: https://github.com/bigbinary/neeto-editor/blob/master/CHANGELOG.md

The logic tries to extract the previous version from the "Unreleased" heading. After switching to commit hashes, this is no longer available. As a result, the changelog updater ends up appending the new entry to the end of the file.

@AbhayVAshokan _a
bug _t

### PR DESCRIPTION
**Pull request by AbhayVAshokan**
Fixes #6 
- Updated logic to insert new changelog entries below the "Unreleased" heading.

@Amaljith-K _a can you please share your thoughts about the same? Once resolved, I'll proceed to fix the tests. 

https://vimeo.com/752862375/fe4b010b6e

### CODE REVIEW

File: app/Actions/PlaceReleaseNotesBelowUnreleasedHeadingAction.php [hunk 1]
```diff
@@ -31,7 +31,8 @@ public function __construct(
      */
     public function execute(Heading $unreleasedHeading, string $latestCommit, string $headingText, string $releaseDate, ?string $releaseNotes, Document $changelog, string $compareUrlTargetRevision): Document
     {
-        $previousVersion = $this->getPreviousVersionFromUnreleasedHeading($unreleasedHeading);
+        // $previousVersion = $this->getPreviousVersionFromUnreleasedHeading($unreleasedHeading);
```
Comment (Amaljith-K) [inline app/Actions/PlaceReleaseNotesBelowUnreleasedHeadingAction.php]: You can remove this line

**Review by Amaljith-K — COMMENTED** [review]
@AbhayVAshokan _a I think it is best to throw an error if an unreleased heading is not found.
If possible make the message beginner friendly. Or otherwise, you can add an explanation on how to place an unreleased heading in the wiki & add a link in the error message.

File: app/Exceptions/UnreleasedHeadingDoesNotExistException.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+namespace App\Exceptions;
+
+use Exception;
+
+class UnreleasedHeadingDoesNotExistException extends Exception
+{
+    public function __construct()
+    {
+        parent::__construct("The 'Unreleased' heading was not found in the CHANGELOG. Please refer to the correct CHANGELOG format at: https://github.com/bigbinary/neeto-editor/blob/master/CHANGELOG.md.");
+    }
```
Comment (AbhayVAshokan) [inline app/Exceptions/UnreleasedHeadingDoesNotExistException.php]: The URL shall be updated with the neetoKB article in neeto-engineering which contains a detailed explanation of how to setup the CHANGELOG file.

**Comment by AbhayVAshokan** [comment]
@Amaljith-K _a please review. These are the new changes.
- Throw an exception when the "Unreleased" heading does not exist ([75e802c](https://github.com/bigbinary/php-changelog-updater/pull/7/commits/75e802c4aed73c1cbef53f003ce446747c92ae1c)) and removed un-reachable code due to this change ([2e3ac75](https://github.com/bigbinary/php-changelog-updater/pull/7/commits/2e3ac751d99bf74b3e24d543899eb9bcd0bdfead)).
- Deleted all irrelevant tests ([c9184e7](https://github.com/bigbinary/php-changelog-updater/pull/7/commits/c9184e74ad5f04a862575b4789da673232238611)).
- Since the new logic directly inserts the entry under the "Unreleased" heading, there should be no content below the "Unreleased" heading and the next heading. Removed all such occurrences across tests ([5197f1f](https://github.com/bigbinary/php-changelog-updater/pull/7/commits/5197f1fe9f98c61466dc03e6acc94688d5093328)).
- `composer build` command ([3c4cc7d](https://github.com/bigbinary/php-changelog-updater/pull/7/commits/3c4cc7d3a2d79e0c13ab4e99b1a5e511c5dca79f)).

**Review by Amaljith-K — APPROVED** [review]
LGTM

**Comment by Amaljith-K** [comment]
@AbhayVAshokan raise a new PR in https://github.com/bigbinary/changelog-updater-action repo with the newly built bundle in it & assign it to me.
We can release that action as `v2` tag and use it like `bigbinary/changelog-updater-action@v2` since it has breaking changes.

**Pull request merged**